test/groovy/BuildDockerAndPublishImageStepTests.groovy

```diff
@@ -355,6 +355,32 @@ class BuildDockerAndPublishImageStepTests extends BaseTest {
     verifyMocks()
   }
 
+  @Test
+  void itBuildsAndDeploysButNoReleaseWhenTriggeredByTagAndSemVerEnabledButNoReleaseDrafter() throws Exception {
+    def script = loadScript(scriptName)
```

Comment:
Need to make [shellMocks respond with an error for the cases](https://github.com/jenkins-infra/pipeline-library/blob/20661392972a865809e5487f2db5fcfaabf1c4fa/test/groovy/BuildDockerAndPublishImageStepTests.groovy#L52-L53) `case {command.contains('gh api ${GH_RELEASES_API_URI}')}:` and `case {command.contains('gh api $env:GH_RELEASES_API_URI')}:` but only for this test. Any idea?

Reply:
You might be interested in checking the last example of https://github.com/jenkinsci/JenkinsPipelineUnit#mocking-shell-steps. let me prepare a suggestion to give hints

Reply:
```suggestion
    def script = loadScript(scriptName)
    
    helper.addShMock('gh api ${GH_RELEASES_API_URI} | jq -e -r \'[ .[] | select(.draft == true and .name == "next").id] | max\'', '[Warn] No release found in GitHub.', 1)
    
    
```

Reply:
Alternative:

```suggestion
    def script = loadScript(scriptName)
    
    helper.registerAllowedMethod('sh', [Map.class], { m -> throw new Exception('[Warn] No release found in GitHub') })
```

Reply:
The whole idea is to override, only for this test, the sh behavior. I assume that you might want to integrate this into the "generic" mock, so you might want to improve the mock structure to handle custom cases such this one: maybe a parameter and you rewrite it?


Don't forget : KISS is not always a good thing.

Reply:
I'm not sure helper.addShMock would have any effect, as the previous ones were overriden with the registerAllowedMethod.

Adding the alternative to the test I also fail to see how other shell scripts would be mocked.
I can copy/paste the complete shellMocks method in this test to return an exception but it doesn't seems clean at all.